app/routes/course-overview.ts

```diff
@@ -73,7 +78,16 @@ export default class CourseOverviewRoute extends BaseRoute {
         });
       }
 
-      return { course };
+      return {
+        course,
+        completedStages: this.#peekCompletedStages(course),
+      };
     }
   }
+
+  #peekCompletedStages(course: CourseModel) {
```

Comment:
Can this be directly picked off `authenticator.currentUser` instead of peeking globally? 

i.e. something like: 

```javascript
this.authenticator.currentUser
  .repositories
  .filter((repository) => repository.course === course)
  .flatMap('courseStageCompletions')
  .mapBy('courseStage')
```

In general we try to avoid `peekAll` usages unless it's a system-wide resource that doesn't belong to a user (like `course`, `concept` etc.). With most user-related resources there's usually a relationship that can get to the resource we want. This also prevents bugs in the future if we pull in course stage completions from other users (it'll end up showing stages as completed when they aren't).

Reply:
I initially tried adding a `@hasMany('courseStageCompletions')` relationship to the `CourseStage` model, and properly specifying the inverse for `@belongsTo('courseStage')` in `CourseStageCompletion` model. This way I could access completions simply via `course.stages.completions`.

But the backend isn't sending `course_stage_completions` for `CourseStage`s and I couldn't figure out the proper `include` string:

```ts
this.store.findAll('course', {
  include: 'extensions,stages,language-configurations.language',
})
```

I'll try use the approach you suggested, but maybe doing it via models/relationships is a better way, @rohitpaulk ?

Reply:
Changed to the suggested method and it's working. Also refactored the model hook a little bit to simplify and de-duplicate the logic.

Reply:
In addition, moved `completedStages` calculation from route's model hook to controller property, as it wasn't properly working in FastBoot mode.